### ISSUE TITLE
Better disclose that CIDER makes network requests to netlify

### ISSUE THREAD

**Issue by ryantate** [clojure-emacs/cider#2956]
**Is your feature request related to a problem? Please describe.**

I never knew CIDER was making silent network requests in the background as I use the REPL. This is important information to me for two reasons:

1. Privacy
2. Never knew I could not (?) use CIDER REPL without network access. Not a limitation I would expec. I am not complaining about that -- but i just imagined it worked by using the local jvm and am glad I did not plan on going to some remote cabin to code clojure :-)

This information is not disclosed in the 42-line "help banner" that displays whenever the REPL starts. It is not disclosed on README.md. The information is mentioned on https://docs.cider.mx/cider/0.26/index.html as "CIDER’s built on top of nREPL, the Clojure networked REPL server... Clojure code gets executed by an nREPL server. CIDER sends requests to the server and processes its responses " 

I argue, if you are going to tell people how to install this software, as you do in the README.md, there could be a line somewhere in this file mentioning that this REPL works via network requests.

Just to give some context, I installed CIDER via melpa (I believe?) after it was recommended in a book. I have been using this for about six months and only realized about the use of Netlify and the network today, when I got an exception due to Netlify returning a 401 http error.

**Describe the solution you'd like**

I would suggest putting a mention of the tool being built on Netlify in the help screen and on README.md. I can make a formal pull request if this is acceptable.

Thank you for CIDER.

**Comment by bbatsov** [comment]
> Just to give some context, I installed CIDER via melpa (I believe?) after it was recommended in a book. I have been using this for about six months and only realized about the use of Netlify and the network today, when I got an exception due to Netlify returning a 401 http error.

The only request it does if to refresh the ClojureDocs local cache (the Netlify service is maintained by me as well) and you have to explicitly request this, so normally noone should encounter the issue you had, unless you're running an older version of CIDER (older versions updated the cache automatically on start, which turned out to be a mistake). 

I've written about the ClojureDocs export in the past (https://metaredux.com/posts/2019/12/14/exporting-clojuredocs-as-edn.html), in case you need more background.

**Comment by ryantate** [comment]
> The only request it does if to refresh the ClojureDocs local cache (the Netlify service is maintained by me as well) and you have to explicitly request this, so normally noone should encounter the issue you had, unless you're running an older version of CIDER (older versions updated the cache automatically on start, which turned out to be a mistake).

Ah, thank you, this would explain it as I am on 0.22 and I started the REPL maybe 20-60 seconds before I saw the first error (ERROR: Unhandled REPL handler exception....). I had time to run eight function invocations before the errors came. At first it seemed to come in response to input but soon it became clear it was just asyncronous errors (presumably it was retrying after the first one failed).

Nice to know I can go to a remote non internet place and code Clojure, if I can ever arrange that :-)

I will close the issue.

Thank you again for CIDER!
